dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SqlViewController.java

```diff
@@ -239,41 +237,34 @@ public WebMessage executeView( @PathVariable( "uid" ) String uid,
     @PostMapping( "/{uid}/refresh" )
     @ResponseBody
     public WebMessage refreshMaterializedView( @PathVariable( "uid" ) String uid )
+        throws NotFoundException,
+        ConflictException
     {
-        SqlView sqlView = sqlViewService.getSqlViewByUid( uid );
-
-        if ( sqlView == null )
-        {
-            return notFound( "SQL view does not exist: " + uid );
-        }
+        SqlView sqlView = getExistingSQLView( uid );
 
         if ( !sqlViewService.refreshMaterializedView( sqlView ) )
         {
-            return conflict( "View could not be refreshed" );
+            throw new ConflictException( "View could not be refreshed" );
```

Comment:
Do we want to add an ErrorCode for this for easier programmatic handling by clients?